nixio/data_frame.py

```diff
@@ -254,18 +254,27 @@ def read_cell(self, position=None, col_name=None, row_idx=None):
             cell = cell[0]
             return cell
 
-    # TODO: allow printing part of the DataFrame
-    def print_table(self):
+    def print_table(self, row_sl=None, col_sl=None):
         """
         Print the whole DataFrame as a table
-        """
-        row_form = "{:^10}" * (len(self.column_names) + 1)
-        print(row_form.format(" ", *self.column_names))
-        if self.units:
-            print(row_form.format("unit", *self.units))
-        for i, row in enumerate(self._h5group.group['data'][:]):
+        :param row_sl: Rows to be printed; None for printing all rows
+        :type row_sl: slice or iterable of int
+        :param col_sl: Columns to be printed; None for printing all columns
+        :type col_sl: slice or iterable of int
+        """
+        if row_sl is None:
+            row_sl = np.s_[:]
+        if col_sl is None:
+            col_sl = np.s_[:]
+        cl = np.array(self.column_names)[col_sl]
+        row_form = "{:^10}" * (len(cl) + 1)
+        print(row_form.format(" ", *cl))
+        if self.units is not None:
+            print(row_form.format("unit", *self.units[col_sl]))
+        for i, row in enumerate(self._read_data(sl=row_sl)[list(cl)]):
             print(row_form.format("Data{}".format(i), *row))
```

Comment:
This is from before this PR, but I don't think the `Data` prefix to the output row numbers is necessary or even very meaningful.  It's also ambiguous: It's the output row index, which is useful, but it can easily be mistaken for the data row index.  Not sure what the best solution here would be.  I can imagine either being useful (or both).